File: App/app/src/main/java/com/cinema/ticket_booking/ui/wallet/WalletViewModel.java
package com.cinema.ticket_booking.ui.wallet;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.cinema.ticket_booking.data.model.response.GiftCardResponse;
import com.cinema.ticket_booking.data.model.response.UserResponse;
import com.cinema.ticket_booking.data.repository.UserRepository;
import com.cinema.ticket_booking.util.Resource;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class WalletViewModel extends ViewModel {

    private final UserRepository userRepo;
    private final MutableLiveData<Resource<UserResponse>> profile = new MutableLiveData<>();
    private final MutableLiveData<Resource<GiftCardResponse>> redeemResult = new MutableLiveData<>();

    @Inject
    public WalletViewModel(UserRepository userRepo) {
        this.userRepo = userRepo;
    }

    public LiveData<Resource<UserResponse>> getProfile() {
        return profile;
    }

    public LiveData<Resource<GiftCardResponse>> getRedeemResult() {
        return redeemResult;
    }

    public void loadProfile() {
        userRepo.getMyProfile().observeForever(profile::setValue);
    }

    public void redeemGiftCard(String code) {
        userRepo.redeemGiftCard(code).observeForever(redeemResult::setValue);
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/ui/wallet/WalletViewModel.kt
package com.cinema.ticket_booking.ui.wallet

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.cinema.ticket_booking.data.model.response.GiftCardResponse
import com.cinema.ticket_booking.data.model.response.UserResponse
import com.cinema.ticket_booking.data.repository.UserRepository
import com.cinema.ticket_booking.util.Resource
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class WalletViewModel @Inject constructor(
    private val userRepo: UserRepository
) : ViewModel() {

    private val _profile = MutableLiveData<Resource<UserResponse>>()
    val profile: LiveData<Resource<UserResponse>> = _profile

    private val _redeemResult = MutableLiveData<Resource<GiftCardResponse>>()
    val redeemResult: LiveData<Resource<GiftCardResponse>> = _redeemResult

    fun loadProfile() {
        userRepo.getMyProfile().observeForever { _profile.value = it }
    }

    fun redeemGiftCard(code: String) {
        userRepo.redeemGiftCard(code).observeForever { _redeemResult.value = it }
    }
}
